### PR TITLE
Adds Categories support to Hub CLI and `-o wide` flag in search command

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
- Adds categories in Info command
  - Adds categories in Search command
      -  Adds `-o wide` flag in search command to display all columns
  
Screenshot from 2021-09-14 10-14-57

Signed-off-by: Puneet Punamiya <ppunamiy@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [x] Run API Unit Tests, Lint Checks, API Design, Golden Files with `make api-check`
- [x] Run UI Unit Tests, Lint Checks with `make ui-check`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: api/pkg/cli/cmd/search/search.go [hunk 1]
```diff
@@ -125,32 +140,41 @@ func (opts *options) run() error {
...
 }
 
 func (opts *options) validate() error {
 
-	if flag.AllEmpty(opts.args, opts.kinds, opts.tags) {
-		return fmt.Errorf("please specify a resource name, --tags or --kinds flag to search")
+	// supportedOutputTypes := sets.NewString("", "wide", "name")
+	// if !supportedOutputTypes.Has(opts.console) {
+	// 	return fmt.Errorf("--output %v is not available", opts.console)
+	// }
```
Comment (vinamra28) [inline api/pkg/cli/cmd/search/search.go:161]: if not needed can you please remove this? :stuck_out_tongue:

**Comment by sm43** [comment]
what if we keep `tags` in normal output? 🤔 using tags it is a bit easy to relate what a resource is about 
eg. docker or test they give what a task is abt..
well category also gives that 😄 but showing one of them would be nice

**Comment by PuneetPunamiya** [comment]
> what if we keep `tags` in normal output?  using tags it is a bit easy to relate what a resource is about
> eg. docker or test they give what a task is abt..
> well category also gives that  but showing one of them would be nice

cc @piyush-garg  @vinamra28  @vdemeester

**Comment by pratap0007** [comment]
https://github.com/tektoncd/hub/blob/main/api/pkg/cli/cmd/search/search.go#L79 ->  we should have categories as well in  short description of search command,right ?

**Pull request merged**